src/Microsoft.AspNetCore.SystemWebAdapters/Caching/Cache.cs

```diff
@@ -61,6 +61,17 @@ private static void AddChangeMonitors(CacheDependency? dependencies, CacheItemPo
 
     public void Insert(string key, object value) => _cache.Set(key, value, new CacheItemPolicy());
 
+    public void Insert(string key, object value, CacheDependency? dependencies)
+    {
+        var policy = new CacheItemPolicy
```

Comment:
What about the dependency argument? The original implementation is here: https://referencesource.microsoft.com/#System.Web/Cache/cache.cs,d083199dcfb5f0da - feel free to copy the implementation from there to here.

Reply:
Is that why the build is failing btw?

Reply:
possibly

Reply:
@twsouthwick `CacheDependency` argument is not currently being consumed in other methods but there is an unused private `AddChangeMonitors` that was introduced in this commit https://github.com/dotnet/systemweb-adapters/commit/66276e5b0cbd7c74d054684b34843f7b9c553127#diff-bfdc712f9e9845bfd447d4eb1cbbafdb48e1d076e57108d882eb3651699290a9R52 -- This is a good potential solution to consume and monitor the cache dependencies. 

I'm happy to refactor other methods as well but I wanted to double check with you first.

Reply:
@twsouthwick I know there was a little bit of code missing from the implementation that I found when I was doing the VirtualPathProvider implementation.
There are changes in my branch https://github.com/CZEMacLeod/systemweb-adapters/tree/czemacleod/vpp-1.1 that I think include this overload and also sort out some other stuff that was not fully implemented.
I've been meaning to complete the branch for VPP and submit a PR, but while I have a good test/sample implementation, I don't have any good explicit tests defined.
It would probably be good if I separated out all the Cache changes from the VPP changes, as they are fixes/enhancements rather than a new feature but I haven't had time to complete this work yet.
Let me see if I can update my branch and pull all the cache specific code out into a PR. 
One thing I found was that there were a few things (like HasChanged) that don't behave quite as logically as you would expect on the System.Web Cache implementation compared to the more generic version that was introduced later. I know that I did catch these inconsistencies and match them like for like in the update I did.

Reply:
I created #351 which contains all the changes I had queued for `Cache` and `CacheDependency`.
I've added the tests from this PR, and the missing overload for `Cache.Insert`. The test here for NoSliding/NoAbsolute is technically covered by [Cache.Insert(String, Object)](https://learn.microsoft.com/en-us/dotnet/api/system.web.caching.cache.insert?view=netframework-4.8.1#system-web-caching-cache-insert(system-string-system-object))
I also have tests when there is an actual dependent object (file or item).